engine/contact/contact_test.go

```diff
@@ -0,0 +1,150 @@
+package contact_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/bitcoin-sv/go-paymail"
+	"github.com/bitcoin-sv/spv-wallet/engine"
+	"github.com/bitcoin-sv/spv-wallet/engine/contact/testabilities"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/tester/fixtures"
+)
+
+func Test_ClientService_AdminCreateContact_Success(t *testing.T) {
+	tests := map[string]struct {
+		contactPaymail   string
+		creatorPaymail   string
+		fullName         string
+		metadata         *engine.Metadata
+		expectedError    error
+		expectedStatus   engine.ContactStatus
+		expectedFullName string
+	}{
+		"Happy path without metadata": {
+			contactPaymail:   fixtures.RecipientExternal.DefaultPaymail(),
+			creatorPaymail:   fixtures.Sender.DefaultPaymail(),
+			fullName:         "John Doe",
+			metadata:         nil,
+			expectedError:    nil,
+			expectedStatus:   engine.ContactNotConfirmed,
+			expectedFullName: "John Doe",
+		},
+		"Happy path with metadata": {
+			contactPaymail: fixtures.RecipientExternal.DefaultPaymail(),
+			creatorPaymail: fixtures.Sender.DefaultPaymail(),
+			fullName:       "John Doe",
+			metadata: &engine.Metadata{
+				"key1": "value1",
+				"key2": 420,
+			},
+			expectedError:    nil,
+			expectedStatus:   engine.ContactNotConfirmed,
+			expectedFullName: "John Doe",
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			//given:
+			given, then := testabilities.New(t)
+
+			service, cleanup := given.Engine()
+			defer cleanup()
+
+			//when:
+			res, err := service.AdminCreateContact(context.Background(), tt.contactPaymail, tt.creatorPaymail, tt.fullName, tt.metadata)
+
+			//then:
+			then.NoError(err).WithResponse(res).WithStatus(tt.expectedStatus).WithFullName(tt.expectedFullName)
```

Comment:
IMHO you should also 
1. check the PubKey - if it matches the one from external paymail server. 
2. And maybe also check the Paymail field that it is paymail of contact
3. and maybe could add an assertion that contact was created for user, and pass the fixtures.Sender and check if OwnerXpubID is matching user.XPubID

to sum up, I would see it like this:
```suggestion
			then.ContactCreated(contact).WithoutError(err)
			  .ForUser(fixtures.Sender) // Maybe it could be named InContactBookOf(fixtures.Sender)
			  .WithContactPaymailOf(fixtures.RecipientExternal)
			  .WithPubKey(fixtures.RecipientExternalPKI)
			  .WithStatus(tt.expectedStatus)
			  .WithFullName(tt.expectedFullName)
```

alternatively it could be something like this:
```suggestion
			then.ContactCreated(contact).WithoutError(err)
			  .ForUser(fixtures.Sender)  // Maybe it could be named InContactBookOf(fixtures.Sender)
			  .ToCounterparty(fixtures.RecipientExternal) // and here we check the paymail and PubKey
			  .WithStatus(tt.expectedStatus)
			  .WithFullName(tt.expectedFullName)
```

Reply:
done :D